# ATE and ROI
This notebook will compute Average Treatment Effect and Return on Investment from running an ad campaign.

## Data Exploration

In [2]:
df <- read.csv("rocketfuel_data.csv")
head(df)

user_id test converted tot_impr mode_impr_day mode_impr_hour
1 1069124 1    0         130      1             20            
2 1119715 1    0          93      2             22            
3 1144181 1    0          21      2             18            
4 1435133 1    0         355      2             10            
5 1015700 1    0         276      5             14            
6 1137664 1    0         734      6             10

In [5]:
# check number of control and test users
sum( df$test == 0 ) # control
sum( df$test == 1 ) # test

# check conversion rates
round( mean( df$converted[ df$test == 0] ) * 100, 2 )
round( mean( df$converted[ df$test == 1] ) * 100, 2 )

[1] 23524

[1] 564577

[1] 1.79

[1] 2.55

## ATE

In [7]:
# compute treatment effect "by hand"
ate <- mean( df$converted[df$test == 1] ) - mean( df$converted[df$test == 0] )
ate_pct <- ate * 100
round(ate_pct, 4)

[1] 0.7692

In [8]:
# compute treatment effect by regression
lm1 <- lm( converted~test, data=df )
summary( lm1 )


Call:
lm(formula = converted ~ test, data = df)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.02555 -0.02555 -0.02555 -0.02555  0.98215 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 0.017854   0.001023   17.46  < 2e-16 ***
test        0.007692   0.001044    7.37  1.7e-13 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.1568 on 588099 degrees of freedom
Multiple R-squared:  9.236e-05,	Adjusted R-squared:  9.066e-05 
F-statistic: 54.32 on 1 and 588099 DF,  p-value: 1.703e-13


The campaign is effective, increasing conversions by 0.7692%.

## ROI

In [11]:
# incremental conversions from campaign
inc_con <- sum( df$test == 1 ) * lm1$coefficients["test"]
round(inc_con, 2)

test 
4342.98

In [13]:
# incremental profit from campaign, averaging $40 per conversion
inc_pft <- inc_con * 40
round(inc_pft, 2)

test 
173719.3

Running the campaign increased conversions by about 4,342.98 and profit by about \$173,719.29

In [16]:
# compute campaign cost, $9 CPM
cost <- sum(df$tot_impr) / 1000 * 9
round(cost, 2)

[1] 131374.6

In [18]:
# compute ROI
ROI <- 100 * (inc_pft - cost) / cost
round(ROI, 2)

test 
32.23

With a campaign cost of \$131,374.64, the return on investment is 32.23%.

In [19]:
# compute additional profit if ad was shown to control group
inc_con2 <- sum( df$test == 0 ) * lm1$coefficients["test"]
inc_pft2 <- inc_con2 * 40
round(inc_pft2, 2)

test 
7238.29

Showing the ad to the control group would increase profit by \$7,238.29